### DISCUSSION TITLE
Any way to get DOM in framework like Observable HQ?

### DISCUSSION THREAD

**Discussion by DevinBayly** [observablehq/framework#1684] [Q&A] [answered]
Hi there,

I was recently working on a force collide framework page and the example I was following made its canvas using `DOM.context2d(width, height)` and I was wondering what the most idiomatic method for this would be in framework? 

I ended up with making the canvas in html below and resizing it to be the width and height that I wanted, but it felt like there might be a more seamless way to translate from the observablehq example.

**Comment by mbostock** [comment, marked as answer]
See the note here: https://observablehq.com/framework/convert#standard-library

> For convenience, we’ve linked to the implementations above so that you can see how they work, and if desired, copy the code into your own Framework app as vanilla JavaScript. For example, for a [2D canvas](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D), you can replace DOM.context2d with:

```js
function context2d(width, height, dpi = devicePixelRatio) {
  const canvas = document.createElement("canvas");
  canvas.width = width * dpi;
  canvas.height = height * dpi;
  canvas.style = `width: ${width}px;`;
  const context = canvas.getContext("2d");
  context.scale(dpi, dpi);
  return context;
}
```

**Reply by DevinBayly** [reply]
Great this totally works for me. Thanks @mbostock !
